models/document-container.js

```diff
@@ -0,0 +1,84 @@
+import EditorDocument from "./editor-document";
+import { sparqlEscapeUri, sparqlEscapeString } from "mu";
+import { querySudo as query } from "@lblod/mu-auth-sudo";
+
+export default class DocumentContainer {
+  /** @type {string} */
+  id;
+  /** @type {string} */
+  uri;
+  /** @type {string} */
+  folder;
+  /** @type {EditorDocument} */
+  currentVersion;
+
+  /**
+   *
+   * @param {{uri: string, id: string, folder: string, currentVersion: EditorDocument}}
+   */
+  constructor({ uri, id, folder, currentVersion }) {
+    this.uri = uri;
+    this.id = id;
+    this.folder = folder;
+    this.currentVersion = currentVersion;
+  }
+
+  static fromBinding(binding) {
+    const currentVersion = new EditorDocument({
+      uri: binding.currentVersion_uri.value,
+      id: binding.currentVersion_id.value,
+      title: binding.currentVersion_title.value,
+      content: binding.currentVersion_content.value,
+    });
+    return new DocumentContainer({
+      uri: binding.uri.value,
+      id: binding.id.value,
+      folder: binding.folder.value,
+      currentVersion,
+    });
+  }
+  /**
+   *
+   * @param {({ uri?: string, id?: string })} queryOptions
+   */
+  static async query(queryOptions) {
+    const { uri, id } = queryOptions;
+    let bindStatement = uri
+      ? `BIND(${sparqlEscapeUri(uri)} AS ?uri)`
+      : id
+        ? `BIND(${sparqlEscapeString(id)} AS ?id)`
+        : "";
```

Comment:
Does it make sense to allow `{}` as queryOptions? While it technically works, it makes more sense to me as an error as we would never want just the first DocumentContainer with no constraints. It would then make the @param `{ uri: string } | { id: string }`, assuming that works in jsdoc.

Reply:
good point, I updated the code accordingly